### PR TITLE
Devtools diaspora post

### PR DESCRIPTION
**Pull request by jennybc**
Made this a separate post, as discussed in #115.

I'm wondering if @jimhester (or @hadley) is willing to carry it over the finish line and co-author?

### CODE REVIEW

File: content/articles/2018-02-devtools-refactor.Rmd [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  url: https://morguefile.com/p/734816
+  author: mconnors
+---
+
+```{r setup, include = FALSE}
+library(usethis)
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
+```
+
+[devtools](https://github.com/r-lib/devtools#readme) is an R package that helps you create, develop, and install R packages. By 2017, it was clear that devtools would be easier to maintain if it was split into a few smaller and more focused packages. For example, there is internal tension from trying to serve two radically opposing aims: a batteries-included framework for rapid iteration in local package development and a minimal-dependency package installer and checker, e.g., for continuous integration settings like [Travis CI](https://travis-ci.org).
```
Comment (batpigandme) [inline content/articles/2018-02-devtools-refactor.Rmd]: The shift between the first and second sentences feels a little off to me. I know we're doing the "what is the package" line, but without an anchoring date the _"By..."_ construct is a little awkward.

File: content/articles/2018-02-devtools-refactor.Rmd [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+Other packages in development on GitHub, but not yet released on CRAN:
+
+  * FILL
+  * THIS
+  * IN?
+
+## Long live devtools!
+
+devtools has a bright future and is an important package for the r-lib/tidyverse team. devtools will continue to exist as a meta-package, exposing functionality from lower-level, domain-specific packages. The implementation will be different from the [tidyverse meta-package](http://tidyverse.tidyverse.org), but the spirit is the same.
```
Comment (batpigandme) [inline content/articles/2018-02-devtools-refactor.Rmd]: If we want to shift away from having r-lib and tidyverse as synonymous (at least in an outward facing way), I would change team ⇨ _teams_.

File: content/articles/2018-02-devtools-refactor.Rmd [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+In steady state, if you use functions like `load_all()` or `install_github()` in your daily workflow, you should attach devtools via `library("devtools")` and carry on.
+
+There are a few reasons you might want to know more about our internal refactoring:
+
+  * The Awkward Adolescent Phase. It is impossible to choreograph the release of a suite of packages on CRAN. Therefore, there may be periods when functionality exists in usethis, for example, that is not yet exposed by devtools. During such times, you could access those specific functions directly through usethis. If you attach devtools in `.Rprofile`, for interactive use, this implies you should attach devtools first and then any of its constituent packages, as they roll out onto CRAN. Again, this should only be necessary during the transition period.
```
Comment (batpigandme) [inline content/articles/2018-02-devtools-refactor.Rmd]: Take out comma after `.Rprofile`?

File: content/articles/2018-02-devtools-refactor.Rmd [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  * If you depend on devtools in production scripts or in a package, you should transition that dependency to the lower level package that hosts the functions you need. More below.
+  * If you find a bug, it is most helpful to report it in the repository of package where that function is defined.
+
+## Don't depend on devtools!
+
+Although devtools is here to stay, we'd prefer to see no new direct reverse dependencies. And once the remaining consituent packages are released, we'd like to see existing dependencies transfer away from devtools when possible. devtools has been a *de facto* meta-package for a while and that will become technically true in the coming months. Like the tidyverse meta-package, devtools is convenient for interactive use, but if you are using specific functions programmatically, i.e. in production scripts or in another package, please do so via the function's primary package home and not through devtools.
```
Comment (batpigandme) [inline content/articles/2018-02-devtools-refactor.Rmd]: * Strike the "And" in favour of "Once the remaining constituent..."?
* Add comma before and in "meta-package for a while*,* and that..."

File: content/articles/2018-02-devtools-refactor.Rmd [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+What are the payoffs for creating these lower-level packages?
+
+  * It becomes easier to update and release devtools and the underlying packages. They have fewer reverse dependencies, especially reverse dependencies via completely unrelated functionality that previously lived "under one roof" for reasons of history.
+  * Client packages have lighter dependencies. You can depend only on the packages that offer the functions you use. 
+  * Finer grained versioning (can depend on pkgbuild 1.2.3 and pkgload 4.5.6 even if devtools uses pkgbuild 4.5.6 and pkgload 4.5.6).
```
Comment (batpigandme) [inline content/articles/2018-02-devtools-refactor.Rmd]: Add _e.g._ before "...can depend on pkgbuild"

**Comment by hadley** [comment]
I think we'll hold off on this until pkgload and pkgbuild are on CRAN (once Jim gets back from vacation)

**Pull request closed without merging**